hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConsts.java

```diff
@@ -343,6 +343,7 @@ private OzoneConsts() {
   public static final String BUCKET_LAYOUT = "bucketLayout";
   public static final String TENANT = "tenant";
   public static final String USER_PREFIX = "userPrefix";
+  public static final String OVERWRITE_GENERATION = "overwriteGeneration";
```

Comment:
Agreed, and similarly for `OmKeyArgs.overwriteGeneration`. Although I haven't taken a look deeply, I think the "generation" concept can be reused in the future. For example, we can ensure follower read adheres to "read-your-own-writes" consistency by passing the generation that was just written to the follower to ensure that the value has been updated in the follower (instead of linearizable read which has higher latency since it requires the follower to contact the leader as well).

Reply:
This constant is only used in the audit log. I think it makes sense to call it overwrite (or probably rewrite) generation to make it distinct from the current key generation. Calling the rewriteGeneration just generation could be confusing if we have a current generation called just "generation" and then this atomic rewrite parameter called "generation".

Reply:
This was meant as a more general comment on the PR which probably wasn't clear. I think there are some places where `rewriteGeneration` makes sense, like the audit log here and the `KeyArgs` that are passed in by the client. Basically the places where the client is instructing the server to use the generation for rewrite purposes. However, when the value is stored or returned like in `OmKeyInfo` and the `KeyInfo` proto I think we should just call it `generation`. At that point the value could technically be used for anything, not just a rewrite.